module/modmanager/manager_test.go

```diff
@@ -1384,33 +1384,49 @@ func TestModularDiscovery(t *testing.T) {
 	err := mgr.Add(ctx, modCfg)
 	test.That(t, err, test.ShouldBeNil)
 
-	// The helper model implements actual (foobar) discovery
+	// The "helper" model implements actual (foobar) discovery
 	reg, ok := resource.LookupRegistration(generic.API, resource.NewModel("rdk", "test", "helper"))
 	test.That(t, ok, test.ShouldBeTrue)
 	test.That(t, reg, test.ShouldNotBeNil)
-
-	// Check that the Discover function is registered and make call
 	test.That(t, reg.Discover, test.ShouldNotBeNil)
-	result, err := reg.Discover(ctx, logger)
-	test.That(t, err, test.ShouldBeNil)
-	t.Log("Discovery result: ", result)
 
-	// Format result
-	jsonData, err := json.Marshal(result)
-	test.That(t, err, test.ShouldBeNil)
-	// Debug: print the JSON data
-	t.Logf("Raw JSON: %s", string(jsonData))
+	testCases := []struct {
```

Comment:
Added testing here to make sure extra can be passed through and change control flow as expected